PythonVisualizations/SimpleSorting.py

```diff
@@ -4,421 +4,23 @@
-                    outline=self.FOUND_COLOR))
-
-                # update the display
-                self.window.update()
-                self.wait(0.1)
-
-                # Animation stops
-                self.highlightCodeTags([], callEnviron)
-                self.cleanUp(callEnviron)
-                return i
-
-            # if not found, then move the index over one cell
-            self.highlightCodeTags('outer_loop_increment', callEnviron)
-            self.moveItemsBy(indexDisplay, (self.CELL_SIZE, 0), sleepTime=0.01)
-            if self.wait(0.01):
-                break
-
-        # Key not found
-        self.highlightCodeTags('key_not_found', callEnviron)
-        self.window.update()
-        self.wait(0.1)
-
-        # Animation stops
-        self.highlightCodeTags([], callEnviron)
-        self.cleanUp(callEnviron)
-        return None
-
-    def toTarget(         # Return the vector between the indexed item's
-            self, index): # current position and its normal position in array
-        return subtract_vector(
-            self.cellCoords(index), 
-            self.canvas.coords(self.list[index].display_shape)[:2])
-    
-    def shuffle(self):
-        self.startAnimations()
-        callEnviron = self.createCallEnvironment()
-
-        nItems = len(self.list)    
-        random.shuffle(self.list)  # Randomly move items internally in array
-
-        coords0 = self.cellCoords(0) # Get the height of a cell
-        height = int(coords0[3] - coords0[1])
-        
-        velocity = [ # Initial velocity is mostly down and towards destination
-            (random.randint(int(dx / 3) - 1, 1) if dx < 0 else
-             random.randint(-1, int(dx / 3) + 1),
-             random.randint(int(height * 1 / 3), height + int(abs(dx) / 8)))
-            for dx, dy in [self.toTarget(i) for i in range(nItems)]]
-        
-        steps = 20
-        for step in range(steps):
-            stepsToGo = steps - step
-            jitter = max(1, int(stepsToGo * 2 / 3))
-            half = max(1, jitter // 2)
-            for i in range(nItems):
-                for item in self.list[i][2:]: # Get drawable shape and val
-                    if item:       # If not None, move it by velocity
-                        self.canvas.move(item, *velocity[i])
-                velocity[i] = add_vector(  # Change velocity to move towards
-                    add_vector(            # target location
-                        multiply_vector(velocity[i], 0.8), 
-                        divide_vector(self.toTarget(i), stepsToGo)),
-                    (random.randint(-jitter, jitter), # Add some random jitter
-                     random.randint(-half, half)))
-            self.wait(0.05)
-
-        # Ensure all items get to new positions
-        self.fixCells()
-
-        # Animation stops
-        self.cleanUp(callEnviron)
+        self.buttons = self.makeButtons()
 
     insertionSortCode = """
 def insertionSort(self):

```

Comment:
If you pause mid insertion sort while items are being shifted over and there is an item in temp, the array contents are wrong because there is two of the item being shifted over

Reply:
I think the visualization accurately shows what's in the array and what's in `temp` throughout the animation.  Here's a snapshot after having copied 59 to `temp` and shifting 94 to the right to take its place:
![Screen Shot 2020-11-12 at 12 18 57 PM](https://user-images.githubusercontent.com/26984827/98991868-4cd1ee00-24e1-11eb-8b43-a0390515a94c.png)
If I resume the animation, the 77 in cell 0 will be shifted to cell 1, replacing the old copy of 94, and leaving 77 in both cells 0 and 1.  On the next step, the 59 in `temp` is copied into cell 0.  The duplicate values in the array during the shifts correctly reflect what is in memory.

If the user stops the animation before the `temp` value is copied back into the array, the duplicate values stick around and whatever was in `temp` is lost.  That's what would happen if the program was interrupted during execution.  That is rather messy conceptually, but it's accurate.